### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by wiggin15**
Various fixes to tests that failed on our environments. I described in the commits what needed fixing.

### CODE REVIEW

File: test/test_psutil.py [hunk 1]
```diff
@@ -3094,18 +3104,13 @@ class TestUnicode(unittest.TestCase):
...
-        if POSIX:
-            st = os.stat(cls.uexe)
-            os.chmod(cls.uexe, st.st_mode | stat.S_IEXEC)
+        cls.uexe = create_temp_executable_file('è')
+        cls.ubasename = os.path.basename(cls.uexe)
+        assert 'è' in cls.ubasename
 
     @classmethod
     def tearDownClass(cls):
-        if not APPVEYOR:
```
Comment (giampaolo) [inline test/test_psutil.py]: mmm this was needed because on APPVEYOR sometimes you get "can't delete file used by another process" errors - I would revert this change

File: test/test_memory_leaks.py [hunk 2]
```diff
@@ -360,13 +360,11 @@ def call(self, function, *args, **kwargs):
...
     def test_cpu_count_logical(self):
-        psutil.cpu_count = psutil._psplatform.cpu_count_logical
-        self.execute('cpu_count')
+        self.execute('cpu_count', logical=True)
 
     @skip_if_linux()
     def test_cpu_count_physical(self):
-        psutil.cpu_count = psutil._psplatform.cpu_count_physical
-        self.execute('cpu_count')
+        self.execute('cpu_count', logical=False)
```
Comment (giampaolo) [inline test/test_memory_leaks.py]: nice

File: test/test_psutil.py [hunk 3]
```diff
@@ -516,6 +516,31 @@ def supports_ipv6():
...
+            f.write("void main() { pause(); }")
+        subprocess.check_call(["gcc", c_file, "-o", path])
+        safe_remove(c_file)
+    else:
+        # fallback - use python's executable
+        shutil.copyfile(sys.executable, path)
+        if POSIX:
+            st = os.stat(path)
+            os.chmod(path, st.st_mode | stat.S_IEXEC)
+    return path
```
Comment (giampaolo) [inline test/test_psutil.py]: nice!

**Comment by wiggin15** [comment]
I reversed the change you asked.
